### ISSUE TITLE
not excute post-link scripts

### ISSUE THREAD

**Issue by nbzu** [prefix-dev/pixi#1573]
### Checks

- [x] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://github.com/prefix-dev/pixi/releases) of pixi, using `pixi --version`.

### Reproducible example

```shell
micromamba activate xxx
micromamba install bioconda::bioconductor-genomeinfodbdata
Rscript -e "library(GenomeInfoDbData)"
```
```shell
pixi project channel add bioconda
pixi add bioconductor-genomeinfodbdata
pixi run Rscript -e "library\(GenomeInfoDbData\)"
```

### Issue description

```shell
Error in library(GenomeInfoDbData) :
  there is no package called ‘GenomeInfoDbData’
Execution halted

```

micromamba works well.
I think pixi not excute the [ post-link scripts ](https://docs.conda.io/projects/conda-build/en/latest/resources/link-scripts.html).

### Expected behavior

excute post-link scripts auto. `library(GenomeInfoDbData)` well.

**Comment by davised** [comment]
I was just coming to post regarding this.

As a user, the workaround can be

```shell
cd project-dir
pixi shell
PREFIX=$CONDA_PREFIX .bioconductor-genomeinfodbdata-post-link.sh
```

I'm going to do this for now.

**Comment by TTTPOB** [comment]
this is still relavent in 2024.10.22, pixi version 0.34.0

**Comment by gohweixun** [comment]
I have just checked this on pixi version 0.36.0, the same issue still persists. Hoping to see a resolution to this, pixi works well in all other workflows.

**Comment by sapetnioc** [comment]
Neither post link nor pre unlink scripts are executed.
The problem is still here for pixi 0.38.0
If it can help, I made a self-content script (pixi being the only requirement) to create and install a minimum package to reproduce the problem.
For the package content, I followed [rattler-build guidelines for packages scripts](https://prefix-dev.github.io/rattler-build/latest/special_files/#post-link-and-pre-unlink-scripts).

```
# Create new test directory
mkdir /tmp/test
cd /tmp/test

# Create a rattler-build recipe with both post link and pre unlink scripts
cat > recipe.yaml << END
package:
  name: pixi-bug-1573
  version: 1.0

build:
  script: |
    mkdir "\$PREFIX/bin"
    echo "touch /tmp/post-link-pixi-bug-1573" > "\$PREFIX/bin/.pixi-bug-1573-post-link.sh"
    echo "touch /tmp/pre-unlink-pixi-bug-1573" >  "\$PREFIX/bin/.pixi-bug-1573-pre-unlink.sh"
    chmod a+rx "\$PREFIX/bin/.pixi-bug-1573-post-link.sh" "\$PREFIX/bin/.pixi-bug-1573-pre-unlink.sh"
END

# Build the package
pixi init
pixi add rattler-build
pixi run rattler-build build

# Install the package
pixi project channel add file:///tmp/test/output
pixi add pixi-bug-1573
```

At the end, two files should exist: `/tmp/post-link-pixi-bug-1573` and `/tmp/pre-unlink-pixi-bug-1573`. But they are not present.

**Comment by looooo** [comment]
is there any plan to support post-link etc. scripts?
Trying to run a post-link.sh I get:
`/.messages.txt: Read-only file system`

**Comment by noamgot** [comment]
+1 - I just encountered this issue with an internal package in my organization :(

**Comment by ruben-arts** [comment]
We're still open to create some kind of `pixi trust` command to allow the post-install scripts of a specific package. https://github.com/prefix-dev/pixi/pull/1585#issuecomment-2216627990

**Comment by ruben-arts** [comment]
That PR didn't implement it.

**Comment by TTTPOB** [comment]
I agree to display a warn if a post/pre link script is skipped.

**Comment by wolfv** [comment]
Yes, in my opinion we should enable the execution of post-link scripts as otherwise parts of the conda ecosystem won't work properly.

**Comment by noamgot** [comment]
Is the fix similar to what was done in https://github.com/prefix-dev/pixi/pull/1585 ?

I tried it locally with a package in my organization with post-link scripts, but it didn't work. 
I have to admit that looking at the scripts themselves, I'm not sure they obey what's defined in [conda's docs](https://docs.conda.io/projects/conda-build/en/stable/resources/link-scripts.html) (e.g., it looks like they don't use `PREFIX` and other environment variables), so maybe I should check differently. If someone can provide an example of how to check this, I can give it a shot.

On a different aspect, as far as I understand, the `with_execute_link_scripts` method is in Rattler, so maybe a warning message should be added on that side.

**Comment by noamgot** [comment]
... this one is fixed by #3347, isn't it? 🥹

**Comment by wolfv** [comment]
Yes - we would love to hear if it works for you now!

**Comment by noamgot** [comment]
Thanks Wolf!
For my particular case (an internal package in my organization) it still doesn't work, but I suspect that's because they don't have a pre-unlink script. I'm sorting this out with the developers.

**Comment by wolfv** [comment]
@noamgot I also need to double check if it already works for `pixi global` or not.
